src/Draco.Compiler/Internal/Semantics/AbstractSyntax/Ast.cs

```diff
@@ -0,0 +1,360 @@
+
+            [Ignore(IgnoreFlags.TransformerAll)]
+            public Type ReturnType => this.DeclarationSymbol.ReturnType;
+        }
+
+        /// <summary>
+        /// A label declaration.
+        /// </summary>
+        public sealed record class Label(
+            [property: Ignore(IgnoreFlags.TransformerTransform)] ParseTree? ParseTree,
+            [property: Ignore(IgnoreFlags.TransformerTransform)] ISymbol.ILabel LabelSymbol) : Decl;
+
+        /// <summary>
+        /// A variable declaration.
+        /// </summary>
+        public sealed record class Variable(
+            [property: Ignore(IgnoreFlags.TransformerTransform)] ParseTree? ParseTree,
+            [property: Ignore(IgnoreFlags.TransformerTransform)] ISymbol.IVariable DeclarationSymbol,
+            Expr? Value) : Decl
+        {
+            [Ignore(IgnoreFlags.TransformerAll)]
+            public Type Type => this.DeclarationSymbol.Type;
+        }
+    }
+
+    /// <summary>
+    /// An expression.
+    /// </summary>
+    public abstract record class Expr : Ast
+    {
+        /// <summary>
+        /// The type this expression evaluates to.
+        /// </summary>
+        [Ignore(IgnoreFlags.TransformerAll)]
+        public abstract Type EvaluationType { get; }
+
+        /// <summary>
+        /// An expression representing unitary value.
+        /// </summary>
+        public record class Unit(
+            [property: Ignore(IgnoreFlags.TransformerTransform)] ParseTree? ParseTree) : Expr
+        {
+            /// <summary>
+            /// A default unit value without a parse tree.
+            /// </summary>
+            public static Unit Default { get; } = new(ParseTree: null);
+
+            [Ignore(IgnoreFlags.TransformerAll)]
+            public override Type EvaluationType => Type.Unit;
+        }
+
+        /// <summary>
+        /// A block expression.
+        /// </summary>
+        public record class Block(
+            [property: Ignore(IgnoreFlags.TransformerTransform)] ParseTree? ParseTree,
+            ImmutableArray<Stmt> Statements,
+            Expr Value) : Expr
+        {
+            [Ignore(IgnoreFlags.TransformerAll)]
+            public override Type EvaluationType => this.Value.EvaluationType;
+        }
+
+        /// <summary>
+        /// A literal expression, i.e. a number, string, boolean value, etc.
+        /// </summary>
+        public sealed record class Literal(
+            [property: Ignore(IgnoreFlags.TransformerTransform)] ParseTree? ParseTree,
+            [property: Ignore(IgnoreFlags.TransformerTransform)] object? Value,
+            [property: Ignore(IgnoreFlags.TransformerTransform)] Type Type) : Expr
+        {
+            [Ignore(IgnoreFlags.TransformerAll)]
+            public override Type EvaluationType { get; } = Type;
+        }
+
+        /// <summary>
+        /// An if-expression with an option elSse clause.
```

Comment:
```suggestion
        /// An if-expression with an optional else clause.
```

Reply:
Why are here and three other doc comments dashes like `if-expression`, but the rest of the doc comments doesn't use these dashes?

Reply:
Good question, likely hack of consistency.